snow/networking/handler/handler.go

```diff
@@ -221,6 +221,9 @@ func (h *handler) Start(ctx context.Context, recoverPanic bool) {
 	h.ctx.Lock.Lock()
 	defer h.ctx.Lock.Unlock()
 
+	// h.asyncMessagePool must be started before async dispatch
+	h.asyncMessagePool.Start()
```

Comment:
@joshua-kim, @StephenButtolph,  I ended up re-introducing the start method.I believe I need it to avoid some goroutines leak in some UTs.
In production code we always call `pool.Start` after its creaton in `snow/networking/handler/handler.go`. However there are some UTs where we don't want to call `handler.Start `since we want to inspect handler queues with Len() (see [TestRouterCrossChainMessages](https://github.com/ava-labs/avalanchego/blob/dev/snow/networking/router/chain_router_test.go#L1244)).
`TestRouterCrossChainMessages` woud leak the pool goroutines if those are started upon `NewPool()`.
Launching goroutines in `handler.Start` solves the issue: goroutines are never started, since `handler.Start` is not called, hence no leak. You can see the fully fixed `TestRouterCrossChainMessages` in my [next PR](https://github.com/ava-labs/avalanchego/pull/1707/files#diff-84a91a02c7606c02cd86085ae012ba8b0d3b756c4dc4c23c352dfbee0878b902R1267)